common/remote/sockfile.cpp

```diff
@@ -6368,13 +6477,32 @@ class CRemoteFileServer : implements IRemoteFileServer, public CInterface
                             break;
                         }
                     }
-                    while (reply.length() < replyLimit);
+                    while (!handleFull(reply, dataStartPos, compressMb, compressor, replyLimit, totalDataSz));
                 }
-                dataLenMarker.write();
+                if (compressor)
+                {
+                    size32_t sz = reply.length()-dataStartPos;
+                    if (sz)
+                    {
+                        totalDataSz += sz;
+                        const void *data = reply.bytes()+dataStartPos;
+                        assertex(compressor->write(data, sz) == sz);
+                        reply.setLength(dataStartPos);
```

Comment:
Do we need a swapWith() here between reply and compressMb ?

Reply:
no it's not done with building up the reply until after the cursor has been written a little bit further past this. Once done it swaps out the completed compressedMb into reply.